extensions/amp-story/1.0/story-analytics.js

```diff
@@ -99,28 +119,66 @@ export class StoryAnalyticsService {
           return;
         }
 
-        this.triggerEvent(AnalyticsEvent.PAGE_VISIBLE);
+        this.triggerEvent(StoryAnalyticsEvent.PAGE_VISIBLE);
 
         const pageIds = this.storeService_.get(StateProperty.PAGE_IDS);
         const pageIndex = this.storeService_.get(
           StateProperty.CURRENT_PAGE_INDEX
         );
         if (pageIndex === pageIds.length - 1) {
-          this.triggerEvent(AnalyticsEvent.LAST_PAGE_VISIBLE);
+          this.triggerEvent(StoryAnalyticsEvent.LAST_PAGE_VISIBLE);
         }
       },
       true /* callToInitialize */
     );
   }
 
   /**
-   * @param {!AnalyticsEvent} eventType
+   * @param {!StoryAnalyticsEvent} eventType
    */
   triggerEvent(eventType) {
-    triggerAnalyticsEvent(
-      this.element_,
-      eventType,
-      this.variableService_.get()
-    );
+    this.incrementPageEventCount_(eventType);
+    triggerAnalyticsEvent(this.element_, eventType, this.updateDetails());
+  }
+
+  /**
+   * Updates event details.
+   * @visibleForTesting
+   * @return {!JsonObject}}
+   */
+  updateDetails() {
+    const vars = this.variableService_.get();
+    const pageId = vars['storyPageId'];
+    let pageEventDetails = {};
+
+    Object.values(StoryAnalyticsEvent).forEach(type => {
```

Comment:
instead of traverse the entire StoryAnalyticsEvent list, can we pass the `eventType` to the `updateDetails()` method. And only pass the `repeated` information for that eventType?

Reply:
@gmajoulet asked to simplify this method in a previous comment (https://github.com/ampproject/amphtml/pull/24634#discussion_r327325739) by removing the `eventType` parameter. @gmajoulet what do you think of @zhouyx 's comment?

Reply:
The main point of my comment was to not store the details as a private property, since it was computed based on the last parameter a method got called with, which is highly unpredictable.

I'm OK with passing the eventType as a parameter as long as we don't store the result within this class.

Reply:
It makes sense to not store the details. If I understand it correctly, information can be found from `this.pageEventCount_`. 

Can we do it like this
`triggerAnalyticsEvent(this.element_, eventType, this.getDetail_(eventType));`
while within the `#this.getDetails_()` method you update and return the `this.pageEventCount_[pageId][eventType]` based on the `eventType` 

@Enriqe @gmajoulet WDYT?

Reply:
Sorry, just came back to this. I updated the PR with what I think will fulfill both requests. LMK if it looks good.